commons/src/main/java/io/aiven/kafka/connect/common/config/AivenCommonConfig.java

```diff
@@ -16,217 +16,35 @@
 
 package io.aiven.kafka.connect.common.config;
 
-import java.time.ZoneId;
-import java.util.ArrayList;
-import java.util.List;
 import java.util.Map;
-import java.util.Objects;
-import java.util.concurrent.TimeUnit;
-import java.util.stream.Collectors;
 
-import org.apache.kafka.common.config.AbstractConfig;
 import org.apache.kafka.common.config.ConfigDef;
-import org.apache.kafka.common.config.ConfigException;
-
-import io.aiven.kafka.connect.common.config.validators.FileCompressionTypeValidator;
-import io.aiven.kafka.connect.common.config.validators.OutputFieldsEncodingValidator;
-import io.aiven.kafka.connect.common.config.validators.OutputFieldsValidator;
-import io.aiven.kafka.connect.common.config.validators.OutputTypeValidator;
-import io.aiven.kafka.connect.common.grouper.RecordGrouperFactory;
-import io.aiven.kafka.connect.common.templating.Template;
-
-public class AivenCommonConfig extends AbstractConfig {
-    public static final String FORMAT_OUTPUT_FIELDS_CONFIG = "format.output.fields";
-    public static final String FORMAT_OUTPUT_FIELDS_VALUE_ENCODING_CONFIG = "format.output.fields.value.encoding";
-    public static final String FORMAT_OUTPUT_TYPE_CONFIG = "format.output.type";
-    public static final String FORMAT_OUTPUT_ENVELOPE_CONFIG = "format.output.envelope";
-    public static final String FILE_COMPRESSION_TYPE_CONFIG = "file.compression.type";
-    public static final String FILE_MAX_RECORDS = "file.max.records";
-    public static final String FILE_NAME_TIMESTAMP_TIMEZONE = "file.name.timestamp.timezone";
-    public static final String FILE_NAME_TIMESTAMP_SOURCE = "file.name.timestamp.source";
-    public static final String FILE_NAME_TEMPLATE_CONFIG = "file.name.template";
-
-    private static final String GROUP_COMPRESSION = "File Compression";
-    private static final String GROUP_FORMAT = "Format";
-    private static final String DEFAULT_FILENAME_TEMPLATE = "{{topic}}-{{partition}}-{{start_offset}}";
-
-    private static final String GROUP_RETRY_BACKOFF_POLICY = "Retry backoff policy";
-    public static final String KAFKA_RETRY_BACKOFF_MS_CONFIG = "kafka.retry.backoff.ms";
+/**
+ * @deprecated Use {@link SinkCommonConfig} instead
+ */
+@Deprecated
+public class AivenCommonConfig extends SinkCommonConfig {
```

Comment:
Shouldn't this extend from CommonConfig, as the AbstractConfig gets all the defaults ?

Reply:
Hi Murali, Unfortunately not, AivenCommonConfig actually only supports SINK type connectors due to private validation methods that require sink specific configuration.

So this class is being deprecated and its functionality is moved to the semantically correct SinkCommonConfig.

Reply:
In my opinion, public interface for connectors is configs in worker.properties, and any changes happening there we could deprecate.

But here, we can already delete this AivenCommonConfig and replace it with SinkCommonConfig, even though it's a public class. No users would be aware of these even with a any kind of release. 

At this moment we have only addKafkaBackoffPolicy, addOutputFieldsFormatConfigGroup, which can be easily moved to SinkCommonConfig, and delete this class.

Reply:
As a public and open source repository we should follow proper deprecation procedures, this would also require a larger change with a larger impact which is not desireable

Reply:
In general I agree for OS repos, if classes are user facing, but in any connectors repo, only worker properties are user facing. 
I am not very much convinced.

Reply:
@Claudenw still not convinced. There would be no developers using this class directly. 
If a property like max.tasks changes to maximum.tasks for ex, I agree we deprecate. 

If we are changing S3Connector name to AivenS3Connector, I agree we deprecate.
But in connectors the scope is zero for someone to use this Config class.

The rule of public classes on OS repos was made for a reason, but I don't think it is always applicable.

I will leave it anyways.